pennylane/compiler/compiler.py

```diff
@@ -41,17 +46,38 @@ class AvailableCompilers:
 
     # The dictionary of installed compiler packages
     # and their entry point loaders.
-    names_entrypoints = {}
+    names_entrypoints = defaultdict(dict)
+
+    # The map consists of supported compiler names (str) and their version compatibility (bool).
+    # This boolean indicates whether the installed version of a compiler package is greater
+    # than or equal to the minimum version.
+    # This value will be updated in `_check_compiler_version` to reduce the required
+    # version checks of installed compiler packages at runtime.
+    names_versions = defaultdict(dict)
+
+
+def _check_compiler_version(name):
+    """Check if the installed version of the given compiler is greater than
+    or equal to the required minimum version.
+    """
+    if AvailableCompilers.names_versions[name]:
+        return  # Used the cached value!
+
+    if name == "catalyst":
+        installed_catalyst_version = metadata.version("pennylane-catalyst")
+        if Version(re.sub(r"\.dev\d+", "", installed_catalyst_version)) < PL_CATALYST_MIN_VERSION:
+            raise CompileError(
+                f"PennyLane-Catalyst {PL_CATALYST_MIN_VERSION} or greater is required, but installed {installed_catalyst_version}"
+            )
+
+    # else
+    AvailableCompilers.names_versions[name] = True
 
 
 def _refresh_compilers():
     """Scan installed PennyLane compiler packages to refresh the compilers
     names and entry points.
     """
-
-    # Refresh the list of compilers
-    AvailableCompilers.names_entrypoints = defaultdict(dict)
-
```

Comment:
Why are these lines removed?

Reply:
It's initialized there, but this line was also resetting it every time. Is that not required?

Reply:
Oooooops! :((